docs/pages/pmd/rules/java/performance.md

```diff
@@ -113,6 +161,152 @@ or AdditiveExpression) and count(.//PrimaryPrefix/Name)=1]
+
+Problem: A Calendar is a heavyweight object and expensive to create. 
+            Solution: Use 'new Date()', Java 8+ java.time.[Local/Zoned]DateTime.now() or joda time '[Local]DateTime.now()'.
+
+**This rule is defined by the following XPath expression:**
+``` xpath
+//PrimaryPrefix[Name[ends-with(@Image, 'Calendar.getInstance')]] [count(../PrimarySuffix) > 2 and ../PrimarySuffix[last()-1][@Image = 'getTime' or @Image='getTimeInMillis']]
+|
+//Block/BlockStatement//Expression/PrimaryExpression/
+PrimaryPrefix/Name[typeIs('java.util.Calendar') and (ends-with(@Image,'.getTime') or ends-with(@Image,'.getTimeInMillis'))]
+|
+//ClassOrInterfaceType[typeIs('org.joda.time.DateTime') or typeIs('org.joda.time.LocalDateTime')][../Arguments/ArgumentList/Expression/PrimaryExpression/PrimaryPrefix/Name[ends-with(@Image, 'Calendar.getInstance')]]
+```
+
+**Example(s):**
+
+``` java
+{%raw%}public class DateStuff {
+    private Date bad1() {
+        return Calendar.getInstance().getTime(); // now
+    }
+    private Date good1a() {
+        return new Date(); // now
+    }
+    private LocalDateTime good1b() {
+        return LocalDateTime.now();
+    }
+    private long bad2() {
+        return Calendar.getInstance().getTimeInMillis();
+    }
+    private long good2() {
+        return System.currentTimeMillis();
+    }
+}{%endraw%}
+```
+
+**Use this rule by referencing it:**
+``` xml
+<rule ref="category/java/performance.xml/AvoidCalendarDateCreation" />
+```
+
+## AvoidConcatInAppend
+
+**Since:** PMD 6.17.0
+
+**Priority:** Medium High (2)
+
+Concatenation of Strings is used inside an StringBuilder.append argument. Problem: Each statement with one or more +-operators creates a hidden temporary StringBuilder, a char[] and a new String object, which all have to be garbage collected.
+            Solution: Use an extra fluent append instead of concatenation.
+
+**This rule is defined by the following XPath expression:**
+``` xpath
+//MethodDeclaration/Block/BlockStatement[
+(.//PrimaryExpression/PrimaryPrefix/Name[substring-after(@Image, '.') = 'append']
+or
+.//PrimaryExpression/PrimarySuffix/@Image = 'append')
+and
+.//PrimaryExpression/PrimarySuffix/Arguments/ArgumentList/Expression[1]/AdditiveExpression[@Image='+'
+and (count(PrimaryExpression/PrimaryPrefix/Name) > 0)
+and not(PrimaryExpression/PrimaryPrefix/Name/@Image=
+ancestor::ClassOrInterfaceBody//FieldDeclaration[@Final='true']//VariableDeclaratorId/@Image)
+and not(PrimaryExpression/PrimaryPrefix/Name/@Image=
+ancestor::Block//LocalVariableDeclaration[@Final='true']//VariableDeclaratorId/@Image)
+]]
+```
+
+**Example(s):**
+
+``` java
+{%raw%}public class StringStuff {
+   private String bad(String arg) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("arg='" + arg + "'");
+        return sb.toString();
+   }
+
+   private String good(String arg) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("arg='").append(arg).append("'");
```

Comment:
The better one:
```java
StringBuilder sb = new StringBuilder("arg='").append(arg).append("'");
```

Perhaps, it's better to use best practices or at least mention that this example isn't the best.

Reply:
Agreed, we should change the sample code. Note: performance.md is generated, the actual change has to happen in performance.xml.

Reply:
> Note: performance.md is generated, the actual change has to happen in performance.xml.

Is it something that can be automated with Danger? :)